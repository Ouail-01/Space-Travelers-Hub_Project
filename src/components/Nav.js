import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../Images/logo.png';
import './Header.css';

const Nav = () => (
  <ul className="nav">
    <li>
      <img src={logo} alt="logo" />
      <h1>Space Traveler&apos;s Hub</h1>
    </li>

    <li>
      <Link className="link" to="/">
        Rockets
      </Link>
    </li>
    <li>
      <Link className="link" to="/Missions">
        Missions
      </Link>
    </li>
    <div className="Line-2" />
    <li>
      <Link className="link" to="/MyProfile">
        My Profile
      </Link>
    </li>
  </ul>
);

export default Nav;
