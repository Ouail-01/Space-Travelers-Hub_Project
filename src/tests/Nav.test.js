import React from 'react';
import Renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Nav from '../components/Nav';

describe('Nav', () => {
  test('renders correctly', () => {
    const tree = Renderer
      .create(
        <Router>
          <Nav />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the correct Rockets page', () => {
    const { getByText } = render(
      <Router>
        <Nav />
      </Router>,
    );
    const rocketsLink = getByText(/Rockets/i);
    fireEvent.click(rocketsLink);
    expect(window.location.pathname).toBe('/');
  });

  test('renders the correct Missions page', () => {
    const { getByText } = render(
      <Router>
        <Nav />
      </Router>,
    );
    const missionsLink = getByText(/Missions/i);
    fireEvent.click(missionsLink);
    expect(window.location.pathname).toBe('/Missions');
  });
});
