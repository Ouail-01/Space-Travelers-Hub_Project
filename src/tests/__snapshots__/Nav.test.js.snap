// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<ul
  className="nav"
>
  <li>
    <img
      alt="logo"
      src="logo.png"
    />
    <h1>
      Space Traveler's Hub
    </h1>
  </li>
  <li>
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      Rockets
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/Missions"
      onClick={[Function]}
    >
      Missions
    </a>
  </li>
  <div
    className="Line-2"
  />
  <li>
    <a
      className="link"
      href="/MyProfile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </li>
</ul>
`;
